inc/saisamplepacket.h

```diff
@@ -47,6 +47,25 @@ typedef enum _sai_samplepacket_type_t
 } sai_samplepacket_type_t;
 
 /**
+  * @brief SAI mode of samplepacket sampling
+  */
+typedef enum _sai_samplepacket_mode_t
+{
+    /** A profile defining sample packet attributes. 
+      * The profile can be configured on a port, in which case, 
+      * an actual session is instantiated which is private for that port. 
+      * In case multiple ports are configured with the same profile, 
+      * sampling is independently done per port. */
+    SAI_SAMPLEPACKET_MODE_EXCLUSIVE,
+    
+    /** An instance of sample packet session. 
+      * This session can be attached to multiple ACL entries. Traffic matching 
```

Comment:
Hi Guohan,

The comment for sample packet mode has been updated.

https://github.com/opencomputeproject/SAI/pull/81/commits

Thanks,

Zhichen

From: lguohan [mailto:notifications@github.com]
Sent: Tuesday, May 03, 2016 8:27 AM
To: opencomputeproject/SAI <SAI@noreply.github.com>
Cc: Zhichen Ren <Zhichenr@mellanox.com>; Author <author@noreply.github.com>
Subject: Re: [opencomputeproject/SAI] Add sample packet mode. (#81)


In inc/saisamplepacket.h<https://github.com/opencomputeproject/SAI/pull/81#discussion_r61901135>:

> @@ -47,6 +47,25 @@ typedef enum _sai_samplepacket_type_t

>  } sai_samplepacket_type_t;

>

>  /**

> +  * @brief SAI mode of samplepacket sampling

> +  */

> +typedef enum _sai_samplepacket_mode_t

> +{

> +    /** A profile defining sample packet attributes.

> +      * The profile can be configured on a port, in which case,

> +      * an actual session is instantiated which is private for that port.

> +      * In case multiple ports are configured with the same profile,

> +      * sampling is independently done per port. */

> +    SAI_SAMPLEPACKET_MODE_EXCLUSIVE,

> +

> +    /** An instance of sample packet session.

> +      * This session can be attached to multiple ACL entries. Traffic matching

By its definition, the mode is not bound to acl rule, it can also be applied to the port. the same is true for the exclusive mode. Can you make this clear in the comments?

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/opencomputeproject/SAI/pull/81/files/0126ebc2b052ca8a8a8c4c09216793d005309ea6#r61901135>